### DISCUSSION TITLE
[Windows] connectorx works in python 3.11 using hex editing

### DISCUSSION THREAD

**Discussion by char101** [sfu-db/connector-x#412] [Ideas]
Hi,

Just want to share my preliminary finding that connectorx 0.3.0 in python 3.10 run with python 3.11 with binary patching

1. rename `connectorx.cp310-win_amd64.pyd` to `connectorx.cp311-win_amd64.pyd`
2. open `connectorx.cp311-win_amd64.pyd` in a hex editor (I recommend [hxd](https://mh-nexus.de/en/hxd/)), search for `python310` and replace with `python311`.

I have only tested it with connectorx 0.3.0, querying from postgresql in Windows 10.

Also looking to the import table, connectorx is almost [abi3](https://docs.python.org/3/c-api/stable.html) compatible minus the `PyUnicode_New` import. It would be good if it can be made abi3 compatible so that it can be used with new python versions without having to release new wheels.

**Comment by Pyroluk** [comment]
Char101, thank you! You saved my aorta from bursting due to yet another python dependency hell.
I took the current 0.3.1 wheel from [here](https://pypi.org/project/connectorx/#files) and patched it according to your instructions.
Works flawlessly on Windows Server 2019 with Python 3.11.0 .

**Comment by fdcastel** [comment]
@char101 Do you have a similar workaround for Python 3.12?

**Comment by char101** [comment]
I think it should be obvious, instead of replacing 10 -> 11, you replace 11 -> 12

![2023-12-12 22_37_12-R__cx_connectorx - XYplorer 24 70 - (Academic)](https://github.com/sfu-db/connector-x/assets/71255/a0f4ac81-4952-412d-8af2-9fe99b593332)
